consensus/hotstuff/committees/consensus_committee.go

```diff
@@ -129,7 +130,63 @@ func (c *Consensus) LeaderForView(view uint64) (flow.Identifier, error) {
 	// This assumption is equivalent to assuming that we build at least one
 	// block in every epoch, which is anyway a requirement for valid epochs.
 	//
-	next := c.state.Final().Epochs().Next()
+	epochs := c.state.Final().Epochs()
+	next := epochs.Next()
+
+	// TMP: EMERGENCY EPOCH CHAIN CONTINUATION [EECC]
+	//
+	// If we reach this code-path, it means we are about to propose or vote
+	// for the first block in the next epoch. If that epoch has not been
+	// committed or set up, rather than stopping consensus, this intervention
+	// will create a new fallback leader selection for the next epoch containing
+	// 6 months worth of views, so that consensus will have leaders specified
+	// for the duration of the current spork, without any epoch transitions.
+	//
+	_, err = next.DKG() // either of the following errors indicates that we have transitioned into EECC
+	if errors.Is(err, protocol.ErrEpochNotCommitted) || errors.Is(err, protocol.ErrNextEpochNotSetup) {
+		current := epochs.Current()
+
+		currentCounter, err := current.Counter()
+		if err != nil {
+			return flow.ZeroID, fmt.Errorf("could not get next epoch currentCounter: %w", err)
+		}
+		identities, err := current.InitialIdentities()
+		if err != nil {
+			return flow.ZeroID, fmt.Errorf("could not get epoch initial identities: %w", err)
+		}
+		// CAUTION: this is re-using the same leader selection seed from the now-ending epoch
+		seed, err := current.Seed(indices.ProtocolConsensusLeaderSelection...)
+		if err != nil {
+			return flow.ZeroID, fmt.Errorf("could not get epoch seed: %w", err)
+		}
+		currentFinalView, err := current.FinalView()
+		if err != nil {
+			return flow.ZeroID, fmt.Errorf("could not get epoch first view: %w", err)
+		}
+
+		// we will inject a fallback leader selection in place of the next epoch
+		counter := currentCounter + 1
+		// the fallback leader selection begins after the final view of the current epoch
+		firstView := currentFinalView + 1
+		selection, err := leader.ComputeLeaderSelectionFromSeed(
+			firstView,
+			seed,
+			int(firstView+leader.EstimatedSixMonthOfViews), // the fallback epoch lasts until the next spork
+			identities.Filter(filter.IsVotingConsensusCommitteeMember),
+		)
+		if err != nil {
+			return flow.ZeroID, fmt.Errorf("could not compute epoch fallback leader selection: %w", err)
+		}
+		c.mu.Lock()
+		c.leaders[counter] = selection
```

Comment:
It will happen exactly once, the first time we attempt to retrieve the leader for a view which falls outside of a committed epoch. The reasoning for why this is true is in https://github.com/onflow/flow-go/pull/1250/files/d31bd893918c750b5a4f35b7d75309c4dd9176bb#diff-7236d580d88b376c2040caf16dec5857f2c23366052f90e6f161d6d4646ca5b3R103-R131